src/analyses/goto_check.cpp

```diff
@@ -844,15 +837,14 @@ void goto_checkt::float_overflow_check(
   {
     // Can overflow if casting from larger
     // to smaller type.
-    assert(expr.operands().size()==1);
-
-    if(expr.op0().type().id() == ID_floatbv)
+    const auto &op = to_typecast_expr(expr).op();
+    if(op.type().id() == ID_floatbv)
     {
       // float-to-float
-      const isinf_exprt op0_inf(expr.op0());
+      const isinf_exprt op_inf(op);
       const isinf_exprt new_inf(expr);
 
-      or_exprt overflow_check(op0_inf, not_exprt(new_inf));
+      or_exprt overflow_check(op_inf, not_exprt(new_inf));
```

Comment:
Nit picking: Do names like "op_inf" and "new_inf" really carry sufficient value to justify those temporaries? If they do, we could actually `std::move` them at least.

Reply:
ok, inlined